### PR TITLE
Use empty blocks instead of nops for empty scopes in IRBuilder

### PR DESCRIPTION
**Pull request by tlively**
When IRBuilder builds an empty non-block scope such as a function body,
an if arm, a try block, etc, it needs to produce some expression to
represent the empty contents. Previously it produced a nop, but change
it to produce an empty block instead. The binary writer and printer have
special logic to elide empty blocks, so this produces smaller output.

Update J2CLOpts to recognize functions containing empty blocks as
trivial to avoid regressing one of its tests.

### CODE REVIEW

File: test/lit/passes/dae_all-features.wast [hunk 1]
```diff
@@ -33,7 +33,8 @@
   ;; CHECK-NEXT:  (local.set $0
   ;; CHECK-NEXT:   (i32.const 1)
   ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT:  (block
+  ;; CHECK-NEXT:  )
```
Comment (kripken) [inline test/lit/passes/dae_all-features.wast:37]: These nested locations surprise me a little. Was the old parser really emitting an empty block there? I am pretty sure it wasn't.

Comment (kripken) [reply test/lit/passes/dae_all-features.wast:37]: I see. Seems reasonable.

Maybe worth checking if say the unoptimized code size tests in Emscripten change with this, just to be safe,
```
./test/runner other.test*unoptimized*
```
(optimized size should not be able to change as empty blocks get removed).

**Review by kripken — APPROVED** [review]

File: test/lit/passes/dae_all-features.wast (hunk 1 above)
Comment (tlively) [reply test/lit/passes/dae_all-features.wast:37]: Hmm, don't those tests skip Binaryen entirely because they are run with `-O0`?

Comment (tlively) [reply test/lit/passes/dae_all-features.wast:37]: Also this only affects the text parser for now.

Comment (kripken) [reply test/lit/passes/dae_all-features.wast:37]: Oh right, nevermind my worries then.

**Pull request merged**